### PR TITLE
test(simulator): Orcaella simulation test suite

### PR DESCRIPTION
**Pull request by asonnino**
## Summary

- Adds \`crates/simulator/tests/orcaella.rs\` with simulation tests covering 6 groups: happy-path at tight configurations, single crash fault survival, partition boundary cases, non-optimal (\`k>0\`) committee sizes, multi-leader sweep, and feasibility error validation
- Relaxes the CFT feasibility bound from \`n >= 5c+1\` to \`n >= 2c+1\` (the classical majority bound) — the stricter bound is only needed for the BFT anti-equivocation condition
- Byzantine equivocation is not injectable with the current simulator — left as a follow-up (would need simulator extension)

## Test groups

| group | scenarios | assertion |
|---|---|---|
| Happy path (k=0, full mesh) | n=3 (f=0,c=1), n=6 (f=1,c=0), n=9 (f=1,c=1) | `assert_progress` |
| OneDown within tolerance | each tight config with one node isolated | `assert_progress` |
| Partition boundary at n=9 (commit=7) | majority=7 (at threshold) and majority=6 (below) | `assert_progress` / `== NoProgress` |
| Non-optimal k>0 at n=20 | crash-only, BFT, mixed | `assert_progress` |
| Multi-leader at n=9 | leader_count ∈ {1, 4} | `assert_progress` |
| Feasibility error | BFT: n=4,f=1,c=0; CFT: n=2,f=0,c=1 | `Err(OrcaellaFaultBoundViolated)` |

## Test plan

- [x] `cargo test -p simulator --test orcaella` — all tests pass
- [x] Pre-commit hook (fmt + clippy + full test suite) passed on all commits

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: crates/simulator/tests/orcaella.rs [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+            ..Default::default()
+        },
+        ..Default::default()
+    });
+    assert_eq!(outcome, Outcome::Pass);
+}
+
+#[test]
+fn happy_path_mixed_n9() {
+    // n=9, f=1, c=1 → commit=7, skip=7, anchor=5 (tight)
```
[the remark this answers is not in the document]
Comment (asonnino) [reply crates/simulator/tests/orcaella.rs]: Fixed in e0cfce7 — comment now reads `anchor=4` (n - 3f - 2c = 9 - 3 - 2 = 4).

File: crates/simulator/tests/orcaella.rs [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+            },
+            ..Default::default()
+        }),
+        10,
+    );
+}
+
+#[test]
+fn happy_path_mixed_n9() {
+    // n=9, f=1, c=1 → commit=7, skip=7, anchor=5 (tight)
```
[the remark this answers is not in the document]
Comment (asonnino) [reply crates/simulator/tests/orcaella.rs]: Same fix — corrected in e0cfce7.

File: crates/simulator/tests/orcaella.rs [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        committee_size: 9,
+        topology: NetworkTopology::Partition(vec![vec![0, 1], vec![2, 3, 4, 5, 6, 7, 8]]),
+        duration_secs: 40,
+        replica_parameters: ReplicaParameters {
+            consensus: orcaella(1, 1, 2),
+            ..Default::default()
+        },
+        ..Default::default()
+    });
+    assert_ne!(outcome, Outcome::Diverged);
```
[the remark this answers is not in the document]
Comment (asonnino) [reply crates/simulator/tests/orcaella.rs]: Already addressed in 31850cd — `partition_at_quorum_n9` now calls `assert_progress(_, 10)` which asserts `Outcome::Pass` + a minimum commit count.

File: crates/simulator/tests/orcaella.rs [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+        committee_size: 9,
+        topology: NetworkTopology::Partition(vec![vec![0, 1, 2], vec![3, 4, 5, 6, 7, 8]]),
+        duration_secs: 20,
+        replica_parameters: ReplicaParameters {
+            consensus: orcaella(1, 1, 2),
+            ..Default::default()
+        },
+        ..Default::default()
+    });
+    assert_ne!(outcome, Outcome::Diverged);
```
[the remark this answers is not in the document]
Comment (asonnino) [reply crates/simulator/tests/orcaella.rs]: Already addressed in 31850cd — `partition_below_quorum_n9` now asserts `assert_eq!(result.outcome, Outcome::NoProgress)`.

File: crates/simulator/tests/orcaella.rs [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+}
+
+fn assert_progress(result: &RunResult<SimulationConfig>, min_commits: u64) {
+    assert_eq!(result.outcome, Outcome::Pass);
+    let commits = SnapshotAggregate::new(&result.metrics).max_committed_leaders();
+    assert!(
+        commits >= min_commits,
+        "expected >= {min_commits} committed leaders, got {commits}"
+    );
+}
```
[the remark this answers is not in the document]
Comment (asonnino) [reply crates/simulator/tests/orcaella.rs:33]: `Outcome::Pass` is set by the simulation runner when all replicas terminate without safety violations — it is not purely leader-count-derived. `max_committed_leaders()` then gives a lower-bound liveness check: at least one replica committed the minimum target. A per-quorum assertion would require re-deriving the commit threshold inside the test harness, which we intentionally keep out of scope here.

File: crates/simulator/tests/orcaella.rs [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+
+#[test]
+fn crash_one_node_n3() {
+    // n=3, f=0, c=1: crash 1 node → 2 active == commit threshold
+    assert_progress(
+        &run(SimulationConfig {
+            committee_size: 3,
+            topology: NetworkTopology::OneDown(0),
+            duration_secs: 30,
```
[the remark this answers is not in the document]
Comment (asonnino) [reply crates/simulator/tests/orcaella.rs:94]: Intentional: `OneDown` simulates exactly what a crash fault looks like at the protocol level (an unreachable node). The tests are named after the *fault model being validated* in Orcaella's parameters (`c` crash faults), not the transport primitive. The distinction between network-isolation and crash-stop does not affect the consensus correctness property being tested.

File: crates/simulator/tests/orcaella.rs [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+
+#[test]
+fn crash_one_node_n6() {
+    // commit=5, 5 active nodes — exactly at threshold
+    assert_progress(
+        &run(SimulationConfig {
+            committee_size: 6,
+            topology: NetworkTopology::OneDown(0),
+            duration_secs: 30,
+            replica_parameters: ReplicaParameters {
```
[the remark this answers is not in the document]
Comment (asonnino) [reply crates/simulator/tests/orcaella.rs:113]: Same reasoning as the sibling comment — the name reflects the Orcaella fault-model parameter (`c` crash faults) being exercised, not the precise network mechanism.

File: crates/simulator/tests/orcaella.rs [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+
+#[test]
+fn crash_one_node_n9() {
+    // commit=7, 8 active nodes — one node to spare
+    assert_progress(
+        &run(SimulationConfig {
+            committee_size: 9,
+            topology: NetworkTopology::OneDown(0),
+            duration_secs: 30,
+            replica_parameters: ReplicaParameters {
```
[the remark this answers is not in the document]
Comment (asonnino) [reply crates/simulator/tests/orcaella.rs:131]: Same reasoning as the sibling comment.

File: crates/simulator/tests/orcaella.rs [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+        "expected >= {min_commits} committed leaders, got {commits}"
+    );
+}
+
+#[test]
+fn happy_path_crash_only_n3() {
+    // n=3, f=0, c=1 → commit=2, skip=3, anchor=1 — CFT minimum (n=2c+1)
+    // Direct skip is disabled (s=n); progress relies on indirect commit.
+    assert_progress(
+        &run(SimulationConfig {
```
[the remark this answers is not in the document]
Comment (asonnino) [reply crates/simulator/tests/orcaella.rs:40]: Valid — the PR description still references n=4. Updating it now.
